app/(dashboard)/[storeId]/(routes)/products/[productId]/components/product-form.tsx

```diff
@@ -46,22 +54,31 @@ const BillboardForm: React.FC<BillboardFormProps> = ({
     const [open, setOpen] = useState(false)
     const [loading, setLoading] = useState(false)
 
-    const title = initialData ? "Edit Billboard" : "Create Billboard";
-    const description = initialData ? "Edit a Billboard" : "Add a new Billboard";
-    const toastMessage = initialData ? "Billboard Updated" : "Billboard Created";
+    const title = initialData ? "Edit Product" : "Create Product";
+    const description = initialData ? "Edit a Product" : "Add a new Product";
+    const toastMessage = initialData ? "Product Updated" : "Product Created";
     const action = initialData ? "Save changes" : "Create";
 
 
-    const form = useForm<BillboardFormValues>({
+    const form = useForm<ProductFormValues>({
         resolver: zodResolver(formSchema),
-        defaultValues: initialData || {
-            label: '',
-            imageUrl: ''
+        defaultValues: initialData ? {
+            ...initialData,
+            price: parseFloat(String(initialData?.price)),
+        } : {
+            name: '',
+            images: [],
+            price: 0,
+            categoryId: '',
+            colorId: "",
+            sizeId: '',
+            isFeatured: false,
+            isArchived: false,
         }
     })
 
```

Comment:
- concepts used here

1. useForm hook:
The useForm hook from the react-hook-form library is used to create a form instance. It provides various utilities and state for managing form state, handling form submission, validation, and other related tasks.

2. resolver and zodResolver:
The resolver option is used to specify the validation resolver for the form. In this case, it's set to zodResolver(formSchema), which means that it's using the Zod schema defined earlier (formSchema) for validating the form inputs.

3. defaultValues:
The defaultValues option is used to set the initial values for the form fields. It's a ternary expression that checks if initialData is truthy. If initialData exists, it spreads its properties into the default values and converts the price property to a float using parseFloat(String(initialData?.price)). If initialData is falsy (null), then default values are provided for the form fields.

Reply:
completed